[Merged by Bors] - feat(logic/basic): add eq_iff_true_of_subsingleton

gebner opened the pull request:
I'm surprised we didn't have this already.
```lean
example (x y : unit) : x = y := by simp
```
---
I'm not sure if this breaks anything.

gebner committed:
feat(logic/basic): add eq_iff_true_of_subsingleton

gebner committed:
Fix linear_algebra/basis

gebner committed:
Make fin_cases work again on (fin 1).

gebner committed:
Fix lucas_lehmer

robertylewis commented on src/tactic/fin_cases.lean:
```diff
@@ -46,7 +46,8 @@ meta def fin_cases_at_aux : Π (with_list : list expr) (e : expr), tactic unit
                         (to_rhs >> conv.interactive.change (to_pexpr h))
         -- Otherwise, call `norm_num`. We let `norm_num` unfold `max` and `min`
         -- because it's helpful for the `interval_cases` tactic.
-        | _ := try $ tactic.interactive.norm_num
```
What if you made this `norm_num1` instead of `norm_num` to avoid calling `simp`? I'm not sure if it's powerful enough for whatever `norm_num` is intended to do here, but maybe it is, and that would be an easier fix.

gebner replied:
I tried this as well, but `interval_cases` seems to heavily rely on the simplifier.  Otherwise you get goals like this:
```
w₂: ⊥ < 2
⊢ ⊥ = 0 ∨ ⊥ = 1
w₂: ⊥ + 1 < 2
⊢ ⊥ + 1 = 0 ∨ ⊥ + 1 = 1
```

Just to recap the issues:
 - `fin_cases` should use `subst` on hypotheses like `i = 0`
 - the new simp lemma simplifies those to `true`
 - however in general we want the full simplifier because of `interval_cases`

gebner replied:
So, I think I've fixed this.  The issue was that I've been using `tactic.interactive.norm_num` in the conv monad, which (in hindsight) obviously doesn't work.  Then I tried `conv.interactive.norm_num`, which also failed because it used the simplifier from the tactic monad.  After fixing `conv.interactive.norm_num`, the tests pass.  I'm keeping my fingers crossed.

gebner committed:
Fix conv.interactive.norm_num

gebner committed:
Fix tidy test.

bryangingechen commented on test/tidy.lean:
```diff
@@ -13,10 +13,7 @@ meta def interactive_simp := `[simp]
-  intro1,
-  induction x,
-  refl
+  simp
```
Should this test be testing something about `tidy` in some way? @semorrison

kim-em replied:
It had been a test about `chain`, checking that it failed when it was meant to fail. The test has become a bad one, because now the underlying tactic is succeeding (yay!). I propose we just delete this test entirely.

jcommelin committed:
Delete old test

To quote @semorrison:
> It had been a test about `chain`, checking that it failed when it was
> meant to fail. The test has become a bad one, because now the underlying
> tactic is succeeding (yay!). I propose we just delete this test
> entirely.

jcommelin committed:
Revert "Delete old test"

This reverts commit 771305d15bbc43abcb1ea8a90c690f70f5fda8cd.

jcommelin committed:
Delete only 1 test
